natsstreaming/nats_streaming.go

```diff
@@ -25,24 +25,40 @@ const (
 	OffsetNewest int64 = -1
 )
 
+// ConnectionTimeOutConfig encapsulates interval and timeout seconds fields
+// used for configuring nats streaming connection loss detection
+type ConnectionTimeOutConfig struct {
+	// Number of seconds to wait for a response from nats streaming
+	Seconds int
+	// Number of attempts to establish a connection
+	Tries int
+}
+
 // AsyncMessageSinkConfig is the configuration parameters for an
 // AsyncMessageSink.
 type AsyncMessageSinkConfig struct {
 	URL       string
 	ClusterID string
 	ClientID  string
 	Subject   string
+	TimeOut   ConnectionTimeOutConfig
```

Comment:
Would be nice to add these parameters to the url initialisation as well.

Reply:
Will Add! :+1: